Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.12.0a7-buster as base
+
+# Install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH /root/.local/bin:$PATH
+
+# Make workspace
+WORKDIR /app
+COPY poetry.toml .
+COPY pyproject.toml .
+
+# Install poetry dependencies
+RUN poetry install
+
+# Copy production files
+FROM base as production
+
+COPY todo_app/templates/layout.html todo_app/templates/
+COPY todo_app/templates/prod_index.html todo_app/templates/index.html
+COPY todo_app/*.py todo_app/
+COPY todo_app/data/*.py todo_app/data/
```

Comment:
I'm a fan of broader copy policies than this - although this does give us the strongest guarantees we won't copy something accidentally, it also means every new file/folder is going to end up needing to change our Dockerfile, and we'll quickly get quite swamped with these!
E.g. carefully checking our `.dockerignore` and swapping this for:
```Dockerfile
COPY todo_app todo_app
```
feels much easier to maintain

(comment about prod-index vs index above)